inhibit/inhibit.go

```diff
@@ -81,14 +84,9 @@ func (ih *Inhibitor) run(ctx context.Context) {
 				level.Error(ih.logger).Log("msg", "Error iterating alerts", "err", err)
 				continue
 			}
-			if a.Resolved() {
-				// As alerts can also time out without an update, we never
-				// handle new resolved alerts but invalidate the cache on read.
-				continue
-			}
-			// Populate the inhibition rules' cache.
+			// Update the inhibition rules' cache.
 			for _, r := range ih.rules {
-				if r.SourceMatchers.Match(a.Labels) {
+				if r.exists(a) || r.SourceMatchers.Match(a.Labels) {
```

Comment:
what is the purpose of `r.exists(a)`? It seems like from the code, if it exists, then it gets set (which doesn't accomplish anything, since it already exists?)

Running the tests without this code still passes.

Reply:
IIUC the only case where `r.exists(a)` would be false is when AlertManager processes a resolved alert that hasn't been seen as firing previously (eg after a restart). Not sure it is worth to keep it then.